modules/dnn/CMakeLists.txt

```diff
@@ -21,14 +21,11 @@ if(OPENCV_DNN_OPENCL AND HAVE_OPENCL)
   add_definitions(-DCV_OCL4DNN=1)
 endif()
 
-if(NOT DEFINED OPENCV_DNN_CUDA AND HAVE_CUDNN AND CUDNN_VERSION VERSION_LESS 8.0)
-  message(STATUS "DNN: CUDNN 8.0 is not supported yes. Details: https://github.com/opencv/opencv/issues/17496")
-endif()
 ocv_option(OPENCV_DNN_CUDA "Build with CUDA support"
     HAVE_CUDA
     AND HAVE_CUBLAS
     AND HAVE_CUDNN
-    AND CUDNN_VERSION VERSION_LESS 8.0
+    AND CUDNN_VERSION
```

Comment:
We can just remove this line (`HAVE_CUDNN` is already here)

Reply:
I have removed `CUDNN_VERSION` line (no idea why I even left it there). Do you want `opencv_dnn: filter out cuda4dnn source code` to be removed too?

Reply:
> to be removed too

No.
This message just shows that condition for `OPENCV_DNN_CUDA` is false.
Removal of `CUDNN_VERSION` fixes that.